merlin/models/torch/transforms/sequences.py

```diff
@@ -338,3 +338,179 @@ def get_seq_length(self, inputs: Dict[str, torch.Tensor]) -> int:
+        Whether to pad sequential inputs before extracting the target(s).
+    max_sequence_length : Optional[int], default=None
+        The maximum length of the sequences after padding.
+        If None, sequences will be padded to the maximum length in the current batch.
+
+    Example usage::
+        batch_output = transform(batch)
+
+        features = {
+            'feature1': torch.tensor([[4, 3], [5, 2]),
+            'feature2': torch.tensor([[3,8], [7,9]])
+        }
+        schema = Schema(["feature1", "feature2"])
+        next_item_op = TabularPredictNext(
+            schema=schema, target='feature1'
+        )
+        transformed_batch = next_item_op(Batch(features))
+    """
+
+    def __init__(
+        self,
+        target: Union[str, List[str], Tags],
+        schema: Optional[Schema] = None,
+        apply_padding: bool = True,
+        max_sequence_length: int = None,
+        name: Optional[str] = None,
+    ):
+        super().__init__(
+            TabularPredictNextModule(
+                schema=schema,
+                target=target,
+                apply_padding=apply_padding,
+                max_sequence_length=max_sequence_length,
+            ),
+            name=name,
+        )
+
+
+class TabularSequenceTransform(nn.Module):
+    """Base PyTorch module for preparing targets from a batch of sequential inputs.
+    Parameters
+    ----------
+    target : Selection
+        The sequential input column that will be used to extract the target.
+        For multiple targets usecase, one should provide a Schema containing
+        all target columns.
+    schema : Optional[Schema]
+        The schema with the sequential columns to be truncated
+    apply_padding : Optional[bool], default=True
+        Whether to pad sequential inputs before extracting the target(s).
+    max_sequence_length : Optional[int], default=None
+        The maximum length of the sequences after padding.
+        If None, sequences will be padded to the maximum length in the current batch.
+    """
+
+    def __init__(
+        self,
+        target: Selection,
+        schema: Optional[Schema] = None,
+        apply_padding: bool = True,
+        max_sequence_length: int = None,
+    ):
+        super().__init__()
+        if schema:
+            self.initialize_from_schema(schema)
+            self._initialized_from_schema = True
+        self.target = select(self.schema, target)
+        self.target_name = self._get_target(self.target)
+        self.padding_idx = 0
+        self.apply_padding = apply_padding
+        if self.apply_padding:
+            self.padding_operator = TabularPadding(
+                schema=self.schema, max_sequence_length=max_sequence_length
+            )
+
+    def initialize_from_schema(self, schema: Schema):
+        self.schema = schema
+        self.features: List[str] = self.schema.column_names
+
+    def _get_target(self, target):
+        return target.column_names
```

Comment:
Shouldn't this be `target.column_names[0]`?

Reply:
The target is expected to be a list of strings, in case there are multiple target features.